ml-proto/test/float32.wasm

```diff
@@ -0,0 +1,52 @@
+(module
+  (func $eq_float32 (param $x f32) (param $y f32) (result i32)
+    (eq.f32 (getlocal $x) (getlocal $y))
+  )
+
+  (func $eq_float64 (param $x f64) (param $y f64) (result i32)
+    (eq.f64 (getlocal $x) (getlocal $y))
+  )
+
+  (func $div_float32 (param $x f32) (param $y f32) (result f32)
+    (div.f32 (getlocal $x) (getlocal $y))
+  )
+
+  (func $div_float64 (param $x f64) (param $y f64) (result f64)
+    (div.f64 (getlocal $x) (getlocal $y))
+  )
+
+  (export "eq_float32" $eq_float32)
+  (export "eq_float64" $eq_float64)
+  (export "div_float32" $div_float32)
+  (export "div_float64" $div_float64)
+)
+
+(asserteq
+  (invoke "eq_float32" (const.f32 1.123456789012345) (const.f32 1.123456789))
```

Comment:
The proposed floating formats just represent the bits of the IEEE754
float32/float64 directly, so all you need at the end is an
`Int64.float_of_bits`.

On Wed, Aug 26, 2015 at 1:01 PM, rossberg-chromium <notifications@github.com
> wrote:

> In ml-proto/test/float32.wasm
> <https://github.com/WebAssembly/spec/pull/29#discussion_r38013736>:
>
> > +  (func $div_float32 (param $x f32) (param $y f32) (result f32)
> > +    (div.f32 (getlocal $x) (getlocal $y))
> > +  )
> > +
> > +  (func $div_float64 (param $x f64) (param $y f64) (result f64)
> > +    (div.f64 (getlocal $x) (getlocal $y))
> > +  )
> > +
> > +  (export "eq_float32" $eq_float32)
> > +  (export "eq_float64" $eq_float64)
> > +  (export "div_float32" $div_float32)
> > +  (export "div_float64" $div_float64)
> > +)
> > +
> > +(asserteq
> > +  (invoke "eq_float32" (const.f32 1.123456789012345) (const.f32 1.123456789))
>
> I agree, it would be good to have the front-end issue errors whenever
> there is rounding. But it seems like that would require implementing our
> own string-to-float conversion. I'm no real expert on that.
>
> At least I changed the tests so that the don't rely on rounding for
> literals.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/WebAssembly/spec/pull/29/files#r38013736>.
>